programs/server/Server.cpp

```diff
@@ -871,6 +874,12 @@ int Server::main(const std::vector<std::string> & /*args*/)
     global_context->setDefaultProfiles(config());
     const Settings & settings = global_context->getSettingsRef();
 
+    if (settings.async_insert_threads)
+        global_context->setAsynchronousInsertQueue(std::make_shared<AsynchronousInsertQueue>(
+            settings.async_insert_threads,
+            settings.async_insert_max_data_size,
+            AsynchronousInsertQueue::Timeout{.busy = settings.async_insert_busy_timeout, .stale = settings.async_insert_stale_timeout}));
```

Comment:
AFAIU right now async queues will be removed after storages shutdown, and this will loose some newly INSERT'ed data (via async INSERT)

Reply:
Do you mean the order of destruction on the whole server shutdown?

Reply:
Yes

Reply:
I'll check the exact impact - but doesn't it the same right now? If we shutdown the server in the middle of insertion, then the new part may not get put properly in many ways - if we don't use WAL in first place.

Reply:
In the middle - yes, no guarantee.
But if the INSERT query finished that it is better to try to flush these data into tables, however right now it will not happens since storages will be already shutted down.